scraper/github/app/utils/github_requests.py

```diff
@@ -3,132 +3,191 @@
+
+            # if we get less than per_page there is no more result
+            if len(items) < per_page:
+                break
+
+            page += 1
+            pagination_limit -= 1  # ensure we respect the pagination
+            # time.sleep(1) # we sleep 1 seconds to no flood the api
+
+        self.logger.info(f"We found a total of {len(users)} users")
+        return users
+
+    def get_user(self, user_name: str) -> Dict:
+        """
+
+        This method is for getting details about one user
+
+        @params : user_name [the username of the dev]
+        @returns : user_info [the dict result form github api]
+
+        """
+
+        self.logger.info(f"Fetching user {user_name}")
+        if len(user_name) <= 2:
+            return {"status": "error", "message": "Please provide a valid username"}
+        # We just remove the default @ if it's provide from the user_name
+        user_name = user_name.replace("@", "") if "@" in user_name else user_name
+
+        # we make a simple request to the api
+
+        r = self.session.get(f"{self.GITHUB_API}/users/{user_name}")
+
+        # We check the status of the request and return a predefined error message
+        schk = self.status_check(r)
         if not schk[0]:
-            if on_pageloaded_error:
-                on_pageloaded_error(schk[1])
             return schk[1]
 
-        # We append or merge
-        items = json.loads(r.content)["items"]
-        if on_pageloaded_success:
-            on_pageloaded_success(items)
-        users = users + items
-    return users
+        # return the result as dict
+        return r.json()
 
+    def get_user_repos(self, user_name: str) -> Union[List[Dict], Dict]:
+        """
 
-def get_user(user_name: str):
-    """
+        This method is for getting details about one user
 
-    This method is for getting details about one user
-
-    @params : user_name [the username of the dev]
-    @returns : user_info [the json result form github api]
-
-    """
+        @params : user_name [the username of the dev]
+        @returns : repositories [the json result form github api]
 
-    if len(user_name) <= 2:
-        return {"status": "error", "message": "Please provide a valid username"}
-    # We just remove the default @ if it's provide from the user_name
-    user_name = user_name.replace("@", "") if "@" in user_name else user_name
+        """
 
-    # we make a simple request to the api
+        self.logger.info(f"Fetching user {user_name} repos")
+        if len(user_name) <= 2:
+            return {"status": "error", "message": "Please provide a valid username"}
+        # We just remove the default @ if it's provide from the user_name
+        user_name = user_name.replace("@", "") if "@" in user_name else user_name
 
-    headers = {"Authorization": "token {}".format(GITHUB_TOKEN)}
-    r = requests.get("{}/users/{}".format(GITHUB_API, user_name), headers=headers)
+        # we make a simple request to the api
```

Comment:
just noticed that this was also unecessary 🙂

Reply:
yeah it is a line from old code.
https://github.com/osscameroon/osscameroon-website/blob/main/scraper/github/app/utils/github_requests.py#L58